test/cli/healthcheck_test.rb

```diff
@@ -15,44 +18,61 @@ class CliHealthcheckTest < CliTestCase
     run_command("perform").tap do |output|
-      assert_match "Health check against /up failed to respond, retrying in 1s...", output
-      assert_match "Health check against /up failed to respond, retrying in 2s...", output
-      assert_match "Health check against /up succeeded with 200 OK!", output
+      assert_match "Health check against /up failed to connect, retrying in 1s...", output
+      assert_match "Health check against /up failed with status 500, retrying in 2s...", output
+      assert_match "Health check against /up failed to run: unknown, retrying in 3s...", output
+      assert_match "Health check succeeded with 200 OK!", output
+    end
+  end
+
+  test "perform failing to run" do
+    # Prevent expected failures from outputting to terminal
+    Thread.report_on_exception = false
+
+    SSHKit::Backend::Abstract.any_instance.stubs(:execute) # No need to execute anything here
+    SSHKit::Backend::Abstract.any_instance.stubs(:capture_with_info)
+      .with(:docker, :run, "--rm", "curlimages/curl:latest", "--silent", "--output", "/dev/null", "--write-out", "'%{http_code}'", "--max-time", "2", "http://localhost:3999/up", raise_on_non_zero_exit: false)
+      .returns("error message")
+    SSHKit::Backend::Abstract.any_instance.stubs(:capture_with_info)
+      .with(:docker, :container, :ls, "--all", "--filter", "name=^healthcheck-app-999$", "--quiet", "|", :xargs, :docker, :logs, "--tail", 50, "2>&1")
+
+    exception = assert_raises SSHKit::Runner::ExecuteError do
+      run_command("perform")
     end
+    assert_equal "Exception while executing on host 1.1.1.1: Health check against /up failed to run: error message", exception.message
   end
 
-  test "perform failing because of curl" do
+  test "perform failing to connect" do
     # Prevent expected failures from outputting to terminal
     Thread.report_on_exception = false
 
     SSHKit::Backend::Abstract.any_instance.stubs(:execute) # No need to execute anything here
     SSHKit::Backend::Abstract.any_instance.stubs(:capture_with_info)
-      .with(:curl, "--silent", "--output", "/dev/null", "--write-out", "'%{http_code}'", "--max-time", "2", "http://localhost:3999/up")
-      .returns("curl: command not found")
+      .with(:docker, :run, "--rm", "curlimages/curl:latest", "--silent", "--output", "/dev/null", "--write-out", "'%{http_code}'", "--max-time", "2", "http://localhost:3999/up", raise_on_non_zero_exit: false)
+      .returns("000")
     SSHKit::Backend::Abstract.any_instance.stubs(:capture_with_info)
       .with(:docker, :container, :ls, "--all", "--filter", "name=^healthcheck-app-999$", "--quiet", "|", :xargs, :docker, :logs, "--tail", 50, "2>&1")
 
     exception = assert_raises SSHKit::Runner::ExecuteError do
       run_command("perform")
     end
-    assert_match "Health check against /up failed to return 200 OK!", exception.message
+    assert_equal "Exception while executing on host 1.1.1.1: Health check against /up failed to connect", exception.message
```

Comment:
We no longer have "missing curl" errors to worry about. Add a test for `000` status, indicating the curl didn't get an HTTP response at all.